### DISCUSSION TITLE
Idea; Show server statuses, player count, Discord intergrations.

### DISCUSSION THREAD

**Discussion by seathasky** [ISpaikI/Enshrouded-Server-Manager#20] [Ideas]
As title says, these things really flesh out server software.

EXAMPLE:

![image](https://github.com/ISpaikI/Enshrouded-Server-Manager/assets/17598024/f57adbc2-034b-4fe9-a3f9-67190e814e13)

Discord alerts would be nice as well for stuff like restarts, scheduled restarts etc.

**Comment by ISpaikI** [comment]
Good idea. We will think about it. Dont know if we have enough knowledge to implement discord, but i can only speak for myself.

**Comment by Crazyloon** [comment]
I've never integrated with Discord, but it sounds like a great idea and a fun learning experience.

Start/Stop notifications would probably be relatively easy, but tracking the number of players connected wouldn't be so easy, because the Enshrouded Server itself doesn't expose any useful information.

**Comment by ISpaikI** [comment]
There could be a workaround with steam web api but we have to figure it out

**Comment by ISpaikI** [comment]
https://stackoverflow.com/questions/56515578/how-to-get-steam-gameserver-player-count-using-steam-api

Dont know if its okay for steam that more than one person uses a steamkey for web api.
Could be against tos.

**Comment by seathasky** [comment]
Possibly. A good place to start would to look at Windowsgsm source. The
screen shot is from my Enshrouded server which uses this software:

https://github.com/WindowsGSM/WindowsGSM

 And this plugin:
https://github.com/ohmcodes/WindowsGSM.Enshrouded

Player status and server status are dynamic so I'm sure they are using
something to check player status as it seems to be working perfectly.

**Comment by ISpaikI** [comment]
Steam web api. But dont know if its ok. Over web api you can ask for informations about the server. Get a file back that gives you all the information that needs to be separated. Thats not that hard but our names are saved on this key.

**Reply by seathasky** [reply]
Yea checking TOS would probably be worthwhile to see if it's okay to do

**Comment by ISpaikI** [comment]
Seathasky is webhook integration enough?
I am coding already on it

**Reply by seathasky** [reply]
I definitely believe webhook is plenty yea

**Reply by ISpaikI** [reply]
Would be with id and token be change able for every server or is one id+token enough?
Will try implement start and stop button, updating and autobackup as a webhook.

**Comment by ISpaikI** [comment]
We getting  into it i think
![esm___Die_Gefallenen_Community_Server_-_Discord_01 02 2024_08_41_17](https://github.com/ISpaikI/Enshrouded-Server-Manager/assets/88611660/0951efec-151d-486e-9949-882a13a60935)

**Comment by seathasky** [comment]
Lovely! Great work!

**Reply by ISpaikI** [reply]
But for now i will add only one link so all server will be in one bot, is that enough?

**Comment by seathasky** [comment]
Yes should be good enough indeed

**Reply by ISpaikI** [reply]
So have some issues get backup and autobackup working. But here a new screenshot of actual build we work on.
![Screenshot_2024-02-01-21-52-08-240_com discord-edit](https://github.com/ISpaikI/Enshrouded-Server-Manager/assets/88611660/86422010-39e7-474c-8466-b412297d8e36)

**Reply by seathasky** [reply]
Lovely 😍

**Comment by ISpaikI** [comment]
So fixed a bug at backup and auto backup, changed colors and messages.
Could be the version will come in the next 24 hours.
![esm___Die_Gefallenen_Community_Server_-_Discord_02 02 2024_11_48_47](https://github.com/ISpaikI/Enshrouded-Server-Manager/assets/88611660/e9b92903-23ae-4d02-bbaa-58c12486d919)

**Reply by ISpaikI** [reply]
And your autobackups could act as a "server alive function" because the process id is checked if running

**Reply by ISpaikI** [reply]
Getting into some troubles if discord setting is not set. Crazyloon is looking into it.

**Comment by ISpaikI** [comment]
Hope you like the new discord feature. We will develop it further in some time.

About the server information we need to talk and get into it.

**Reply by seathasky** [reply]
It's perfect, thank you for adding
